homeassistant/components/climate/honeywell.py

```diff
@@ -210,6 +212,24 @@ def update(self):
             self._name = data['name']
             self._is_dhw = False
 
+        if self.entity_id is None:
+            self.entity_id = async_generate_entity_id(
+                ENTITY_ID_FORMAT, self.name, hass=self.hass)
+
+        # The underlying library doesn't expose the thermostat's mode
+        # but we can pull it out of the big dictionary of information.
+
+        device_data = list(
+            filter(lambda x: x['name'] == self.name,
+                   self.device.full_data['devices']))
+
+        if device_data:
+            self.device.system_mode = (
+                device_data[0]['thermostat']
+                ['changeableValues']['mode'])
+
+        self.hass.async_add_job(self.async_update_ha_state())
```

Comment:
Remove this. When home assistant state is updated after a service call, `update` (this method) on entities will be called automatically for polling entities.